encoding/marshal.go

```diff
@@ -228,6 +228,16 @@ func (d *Decoder) DecodeAll(vs ...interface{}) error {
 
 // readN reads n bytes and panics if the read fails.
 func (d *Decoder) readN(n int) []byte {
+	if buf, ok := d.r.(*bytes.Buffer); ok {
+		b := buf.Next(n)
+		if len(b) != n {
+			panic(io.ErrUnexpectedEOF)
+		}
+		if d.n += n; d.n > maxDecodeLen {
```

Comment:
Was this check already in place elsewhere?

Reply:
yep, it's checked in the decoder's `Read` method, which was called by `readN`